dio/lib/src/dio_event_processor.dart

```diff
@@ -2,25 +2,16 @@ import 'dart:async';
 
 import 'package:dio/dio.dart';
 import 'package:sentry/sentry.dart';
-// ignore: implementation_imports
-import 'package:sentry/src/sentry_exception_factory.dart';
 
 /// This is an [EventProcessor], which improves crash reports of [DioError]s.
 /// It adds information about [DioError.requestOptions] if present and also about
 /// the inner exceptions.
 class DioEventProcessor implements EventProcessor {
-  // Because of obfuscation, we need to dynamically get the name
-  static final _dioErrorType = (DioError).toString();
-
   /// This is an [EventProcessor], which improves crash reports of [DioError]s.
   DioEventProcessor(this._options);
 
   final SentryOptions _options;
 
-  SentryExceptionFactory get _sentryExceptionFactory =>
-      // ignore: invalid_use_of_internal_member
-      _options.exceptionFactory;
-
   @override
   FutureOr<SentryEvent?> apply(SentryEvent event, {Hint? hint}) {
     final dynamic dioError = event.throwable;
```

Comment:
I guess this should be done for the first (or last depending on the order of the exceptions) `DioError` in `event.exceptions`.

Reply:
Since we are enriching the event, and this throwable is the root one (first in the exceptions list), i think it's ok to access it this way.

We could think about removal, as the same instance is now also present on the first `SentryException`, but i'm not aware of all the implications of such a change.

Reply:
> Since we are enriching the event, and this throwable is the root one (first in the exceptions list), i think it's ok to access it this way.

I'm pro access via the list, since the whole idea of this change is, that this should also work if the `DioError` is not the root exception.

Reply:
Done 👍